generated/nifake/nifake/unit_tests/test_session.py

```diff
@@ -1,116 +1,81 @@
 import array
-import ctypes
 import datetime
 import hightime
-import math
 import nifake
 import nifake.errors
 import numpy
-import platform
-import warnings
 
 from unittest.mock import MagicMock
 from unittest.mock import patch
 
-import _matchers
 import _mock_helper
 
-# from unittest.mock import ANY
-# Tests
-
-
 SESSION_NUM_FOR_TEST = 42
 
 
 class TestSession(object):
 
-    class PatchedLibrary(nifake._library.Library):
-        def __init__(self, ctypes_library):
-            super().__init__(ctypes_library)
-
+    class PatchedLibraryInterpreter(nifake._library_interpreter.LibraryInterpreter):
+        def __init__(self, encoding):
             for f in dir(self):
-                if f.startswith("niFake_") and not f.endswith("_cfunc"):
-                    setattr(self, f, MagicMock())
+                if not f.startswith("_"):
+                    setattr(self, f, MagicMock(spec_set=getattr(self, f), side_effect=_mock_helper.MockFunctionCallError(f)))
 
     def setup_method(self, method):
-        self.patched_library = self.PatchedLibrary(None)
-        self.patched_library_singleton_get = patch('nifake._library_interpreter._library_singleton.get', return_value=self.patched_library)
+        self.patched_library_interpreter = self.PatchedLibraryInterpreter(None)
+        self.patched_library_singleton_get = patch('nifake.session._library_interpreter.LibraryInterpreter', return_value=self.patched_library_interpreter)
```

Comment:
This name `self.patched_library_singleton_get` seems wrong, now.

Reply:
Renamed to `patched_lirary_interpreter_ctor`. (I'll push the fix when I update the dmm/dcpower metadata again.)